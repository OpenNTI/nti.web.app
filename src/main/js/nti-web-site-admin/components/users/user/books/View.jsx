import React from 'react';
import PropTypes from 'prop-types';
import {Loading} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';
import {LinkTo} from '@nti/web-routing';

import ErrorMessage from '../../../common/ErrorMessage';
import Card from '../../../common/Card';

import BookListItem from './BookListItem';
import Store from './Store';

const DEFAULT_TEXT = {
	error: 'Unable to load transcript.'
};
const t = scoped('nti-site-admin.users.user.Transcript', DEFAULT_TEXT);

const propMap = {
	items: 'items',
	loading: 'loading',
	error: 'error',
};

@Store.connect(propMap)
export default class SiteAdminUserBooks extends React.Component {
	static propTypes = {
		user: PropTypes.object,
		store: PropTypes.object,

		items: PropTypes.array,
		loading: PropTypes.bool,
		error: PropTypes.any,
	}

	get store () {
		return this.props.store;
	}


	componentDidMount () {
		const {user} = this.props;

		this.store.loadBooks(user);
	}


	componentWillUnmount () {
		const {user} = this.props;

		this.store.unloadBooks(user);
	}


	componentDidUpdate (oldProps) {
		const {user:newUser} = this.props;
		const {user: oldUser} = oldProps;

		if (newUser !== oldUser) {
			this.store.loadBooks(newUser);
		}
	}


	render () {
		const {loading, error} = this.props;

		return (
			<div className="site-admin-user-transcripts">
				{loading && (<Loading.Mask />)}
				{!loading && this.renderItems()}
				{error && (<ErrorMessage>{t('error')}</ErrorMessage>)}
			</div>
		);
	}

	renderItems () {
		const {items} = this.props;

		if (!items || !items.length) { return this.renderEmptyState(); }

		return (
			<Card>
				<ul>
					{items.map((item, index) => {
						return (
							<li key={index}>
								<LinkTo.Object object={item} context="site-admin.users.user-books.list">
									<BookListItem book={item.Bundle} />
								</LinkTo.Object>
							</li>
						);
					})}
				</ul>
			</Card>
		);
	}


	renderEmptyState () {

	}
}
