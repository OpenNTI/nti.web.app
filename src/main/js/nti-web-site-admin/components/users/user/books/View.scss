@import '~@nti/style-common';

.site-admin-user-transcripts {
	min-height: 200px;
	position: relative;

	ul {
		li:not(:last-of-type) {
			box-shadow: 0 1px 0 0 var(--border-grey-light);
		}

		li a {
			text-decoration: none;
		}
	}
}
