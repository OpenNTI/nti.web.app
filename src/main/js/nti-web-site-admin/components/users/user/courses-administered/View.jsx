import { useEffect } from 'react';

import { scoped } from '@nti/lib-locale';
import { EmptyState, List, Loading } from '@nti/web-commons';
import { LinkTo } from '@nti/web-routing';
import { EnrollmentListItem } from '@nti/web-course';

import ErrorMessage from '../../../common/ErrorMessage';
import Card from '../../../common/Card';

import Store from './Store';

const t = scoped('nti-site-admin.users.user.courses-administered.View', {
	error: 'Unable to load courses.',
	noCourses: 'This user is not administering any courses',
});

function SiteAdminUserAdministeredCourses({ user }) {
	const { loading, error, load, unload } = Store.useValue();

	useEffect(() => {
		load(user);
		return () => {
			unload(user);
		};
	}, [user]);

	const onEnrollmentChange = () => {
		load(user);
	};

	return (
		<div
			css={css`
				margin-top: 54px;
				min-height: 200px;
				position: relative;
			`}
		>
			{loading ? (
				<Loading.Mask />
			) : (
				<>
					<Items {...{ user, onEnrollmentChange }} />
				</>
			)}
			{error && <ErrorMessage>{t('error')}</ErrorMessage>}
		</div>
	);
}

export default Store.compose(SiteAdminUserAdministeredCourses);

const ListItem = styled.li`
	&:not(:last-of-type) {
		box-shadow: 0 1px 0 0 var(--border-grey-light);
	}

	a {
		text-decoration: none;
	}
`;

function Items({ onEnrollmentChange }) {
	const { items } = Store.useValue();

	if (!items?.length) {
		return <EmptyState>{t('noCourses')}</EmptyState>;
	}

	return (
		<Card>
			<List.Unadorned>
				{items.map((item, index) => (
					<ListItem key={index}>
						<LinkTo.Object
							object={item}
							context="site-admin.users.user-administered-courses.list"
						>
							<EnrollmentListItem enrollment={item} />
						</LinkTo.Object>
					</ListItem>
				))}
			</List.Unadorned>
		</Card>
	);
}
