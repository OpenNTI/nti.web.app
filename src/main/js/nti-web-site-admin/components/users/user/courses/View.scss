@import '~@nti/style-common';

.site-admin-user-transcripts {
	min-height: 200px;
	position: relative;

	.user-transcript-controls {
		display: flex;
		flex-direction: row;
		justify-content: flex-end;
		margin-bottom: 1rem;
	}

	ul {
		li:not(:last-of-type) {
			box-shadow: 0 1px 0 0 var(--border-grey-light);
		}

		li a {
			text-decoration: none;
		}
	}
}
