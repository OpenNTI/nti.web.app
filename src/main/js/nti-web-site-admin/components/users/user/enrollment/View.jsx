import React from 'react';

export default function UserEnrollment () {
	return (
		<div>
			User Enrollment
		</div>
	);
}
