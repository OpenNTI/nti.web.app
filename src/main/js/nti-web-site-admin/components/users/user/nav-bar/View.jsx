
import Card from '../../../common/Card';

import Identity from './Identity';
import Tabs from './Tabs';
import EmailUpdate from './EmailUpdate';

export default function SiteAdminUserIdentity(props) {
	return (
		<>
			<Card className="site-admin-user-nav-bar">
				<Identity {...props} />
				<Tabs {...props} />
			</Card>
			<EmailUpdate {...props} />
		</>
	);
}
