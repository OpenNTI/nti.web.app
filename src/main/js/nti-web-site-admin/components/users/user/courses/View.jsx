import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';

import { List, Loading, Button } from '@nti/web-commons';
import { decorate } from '@nti/lib-commons';
import { scoped } from '@nti/lib-locale';
import { LinkTo } from '@nti/web-routing';
import { EnrollmentListItem, Enrollment } from '@nti/web-course';

import ErrorMessage from '../../../common/ErrorMessage';
import Card from '../../../common/Card';

import Store from './Store';

const DEFAULT_TEXT = {
	error: 'Unable to load courses.',
	noCourses: 'This user is not enrolled in any courses',
	manage: 'Manage Courses',
};
const t = scoped('nti-site-admin.users.user.courses.View', DEFAULT_TEXT);

const propMap = {
	items: 'items',
	loading: 'loading',
	error: 'error',
};

class SiteAdminUserCourses extends React.Component {
	static propTypes = {
		user: PropTypes.object,
		store: PropTypes.object,

		items: PropTypes.array,
		loading: PropTypes.bool,
		error: PropTypes.any,
	};

	get store() {
		return this.props.store;
	}

	componentDidMount() {
		const { user } = this.props;

		this.store.loadTranscript(user);
	}

	componentWillUnmount() {
		const { user } = this.props;

		this.store.unloadTranscript(user);
	}

	componentDidUpdate(prevProps) {
		const { user: newUser } = this.props;
		const { user: oldUser } = prevProps;

		if (newUser !== oldUser) {
			this.store.loadTranscript(newUser);
		}
	}

	onEnrollmentChange = () => {
		const { user } = this.props;

		this.store.loadTranscript(user);
	};

	render() {
		const { loading, error } = this.props;

		return (
			<div className="site-admin-user-transcripts">
				{loading && <Loading.Mask />}
				{!loading && this.renderControls()}
				{!loading && this.renderItems()}
				{error && <ErrorMessage>{t('error')}</ErrorMessage>}
			</div>
		);
	}

	renderControls() {
		const { user } = this.props;

		if (!user.hasLink('EnrollUser')) {
			return null;
		}

		return (
			<div className="user-transcript-controls">
				<Enrollment.Admin.Prompt.Trigger
					user={user}
					onChange={this.onEnrollmentChange}
				>
					<Button rounded>{t('manage')}</Button>
				</Enrollment.Admin.Prompt.Trigger>
			</div>
		);
	}

	renderItems() {
		const { items } = this.props;

		if (!items || !items.length) {
			return this.renderEmptyState();
		}

		return (
			<Card>
				<List.Unadorned>
					{items.map((item, index) => {
						return (
							<li key={index}>
								<LinkTo.Object
									object={item}
									context="site-admin.users.user-transcript.list"
								>
									<EnrollmentListItem
										enrollment={item}
										onChange={this.onEnrollmentChange}
									/>
								</LinkTo.Object>
							</li>
						);
					})}
				</List.Unadorned>
			</Card>
		);
	}

	renderEmptyState() {
		return <div className="empty-state">{t('noCourses')}</div>;
	}
}

export default decorate(SiteAdminUserCourses, [Store.connect(propMap)]);
