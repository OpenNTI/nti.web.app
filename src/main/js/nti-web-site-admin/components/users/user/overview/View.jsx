import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { Widgets } from '@nti/web-reports';
import { getService } from '@nti/web-client';

import DateValue from '../../../common/DateValue';

import ActiveTimes from './ActiveTimes';
import RecentSessions from './RecentSessions';
import LastLogin from './LastLogin';

const { ActiveDays } = Widgets;

const HISTORICAL_SESSIONS_LINK = 'HistoricalSessions';

const NUM_SESSIONS_TO_SHOW = 5;

export default class SiteAdminUserOverview extends React.Component {
	static propTypes = {
		user: PropTypes.object,
	};

	constructor(props) {
		super(props);

		this.state = { loading: true };
	}

	componentDidMount() {
		this.loadData();
	}

	async loadData() {
		const { user } = this.props;

		try {
			const service = await getService();
			const historicalSessions = user.getLink(HISTORICAL_SESSIONS_LINK);
			const results = historicalSessions
				? await service.get(historicalSessions)
				: {};
			const items = (results.Items || []).slice(0, NUM_SESSIONS_TO_SHOW);

			this.setState({ loading: false, historicalSessions: items });
		} catch (e) {
			this.setState({ loading: false, historicalSessions: [] });
		}
	}

	render() {
		const { user } = this.props;
		const { loading, historicalSessions } = this.state;

		return (
			<div className="site-admin-user-overview">
				<div className="joined">
					<DateValue date={user.getCreatedTime()} label="Joined" />
				</div>
				<div className="last-login">
					<LastLogin
						user={user}
						loading={loading}
						historicalSessions={historicalSessions}
					/>
				</div>
				<div className="activity-by-day">
					<ActiveDays entity={user} />
				</div>
				<div className="active-times">
					<ActiveTimes user={user} />
				</div>
				<div className="recent-sessions">
					<RecentSessions
						user={user}
						loading={loading}
						historicalSessions={historicalSessions}
					/>
				</div>
			</div>
		);
	}
}
