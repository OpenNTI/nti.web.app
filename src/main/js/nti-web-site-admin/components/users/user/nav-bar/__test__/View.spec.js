import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import View from '../View';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = () => {
	return {};
};

const onBefore = () => {
	jest.useFakeTimers();
	setupTestClient(getMockService());
};

const onAfter = () => {
	tearDownTestClient();
};

const getUser = hasTranscriptLink => {
	return {
		Username: 'testUser',
		getID: () => 'testUser',
		hasLink: () => hasTranscriptLink,
		email: 'testUser@test.com',
	};
};

/* eslint-env jest */
describe('Site admin user info nav bar test', () => {
	beforeEach(() => onBefore());
	afterEach(onAfter);

	test('Basic render test (has transcript link)', async () => {
		const user = getUser(true);

		const cmp = renderer.create(<View user={user} />);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Basic render test (has no transcript link)', async () => {
		const user = getUser(false);

		const cmp = renderer.create(<View user={user} />);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
