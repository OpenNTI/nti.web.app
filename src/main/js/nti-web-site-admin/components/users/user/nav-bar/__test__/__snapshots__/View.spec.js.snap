// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user info nav bar test Basic render test (has no transcript link) 1`] = `
<div
  className="site-admin-card site-admin-user-nav-bar"
>
  <div
    className="site-admin-user-identity"
  >
    <div
      className="actions"
    >
      <span
        className="email"
      >
        <a
          href="mailto:testUser@test.com"
        >
          Email
        </a>
      </span>
      <div
        className="nti-link-to-path profile"
        to="#"
      >
        <i
          className="icon-goto"
        />
        <span>
          Profile
        </span>
      </div>
    </div>
  </div>
  <div
    className="site-admin-tabs-container"
  >
    <ul
      className="unadorned-list site-admin-tabs"
    >
      <li>
        <div
          className="nti-link-to-path"
          exact={true}
          to="./"
        >
          Overview
        </div>
      </li>
      <li />
      <li />
      <li />
      <li>
        <div
          className="nti-link-to-path"
          to="./reports"
        >
          Reports
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Site admin user info nav bar test Basic render test (has transcript link) 1`] = `
<div
  className="site-admin-card site-admin-user-nav-bar"
>
  <div
    className="site-admin-user-identity"
  >
    <div
      className="actions"
    >
      <span
        className="email"
      >
        <a
          href="mailto:testUser@test.com"
        >
          Email
        </a>
      </span>
      <div
        className="nti-link-to-path profile"
        to="#"
      >
        <i
          className="icon-goto"
        />
        <span>
          Profile
        </span>
      </div>
    </div>
  </div>
  <div
    className="site-admin-tabs-container"
  >
    <ul
      className="unadorned-list site-admin-tabs"
    >
      <li>
        <div
          className="nti-link-to-path"
          exact={true}
          to="./"
        >
          Overview
        </div>
      </li>
      <li />
      <li />
      <li>
        <div
          className="nti-link-to-path"
          to="./transcript"
        >
          Transcript
        </div>
      </li>
      <li>
        <div
          className="nti-link-to-path"
          to="./reports"
        >
          Reports
        </div>
      </li>
    </ul>
  </div>
</div>
`;
